goal_src/jak2/levels/common/enemy/guards/crimson-guard-level.gc

```diff
@@ -7,3 +7,4144 @@
+      (set! (-> v1-2 nav callback-info) *nav-enemy-null-callback-info*)
+      )
+    0
+    (let ((v1-5 self))
+      (set! (-> v1-5 enemy-flags) (the-as enemy-flag (logclear (-> v1-5 enemy-flags) (enemy-flag enemy-flag37))))
+      )
+    0
+    (none)
+    )
+  :exit (behavior ()
+    (let ((v1-1 (-> self root-override2 root-prim)))
+      (logclear! (-> (the-as collide-shape-prim-group v1-1) child 0 prim-core action) (collide-action no-standon))
+      (logior! (-> (the-as collide-shape-prim-group v1-1) child 1 prim-core action) (collide-action no-standon))
+      (set! (-> (the-as collide-shape-prim-group v1-1) child 1 prim-core collide-as) (collide-spec enemy))
+      (set! (-> (the-as collide-shape-prim-group v1-1) child 1 prim-core collide-with)
+            (collide-spec backgnd jak obstacle hit-by-others-list player-list pusher)
+            )
+      )
+    (none)
+    )
+  :trans (behavior ()
+    '()
+    (none)
+    )
+  :code (behavior ()
+    (ja-channel-push! 1 (seconds 0.2))
+    (ja-no-eval :group! (-> self draw art-group data (-> self anim-get-up-back))
+                :num! (seek!
+                  (the float
+                       (+ (-> (the-as art-joint-anim (-> self draw art-group data (-> self anim-get-up-back))) frames num-frames) -1)
+                       )
+                  )
+                :frame-num 0.0
+                )
+    (until (ja-done? 0)
+      (suspend)
+      (ja :num! (seek!))
+      )
+    (set! (-> self hit-face) (the-as uint -1))
+    (go-hostile self)
+    (none)
+    )
+  :post (behavior ()
+    (let ((a0-0 (-> self nav state))
+          (v1-1 *null-vector*)
+          )
+      (logior! (-> a0-0 flags) (nav-state-flag directional-mode))
+      (set! (-> a0-0 travel quad) (-> v1-1 quad))
+      )
+    0
+    (enemy-method-55 self)
+    (none)
+    )
+  )
+
+;; WARN: Return type mismatch object vs none.
+(defmethod crimson-guard-level-method-204 crimson-guard-level ((obj crimson-guard-level))
+  (let ((s4-0 (vector-x-quaternion! (new 'stack-no-clear 'vector) (-> obj root-override2 quat)))
+        (s5-0 (-> obj root-override2))
+        (s3-0 (lambda ((arg0 crimson-guard-level) (arg1 collide-shape-moving) (arg2 vector))
+                (let ((s4-0 (new 'stack-no-clear 'vector))
+                      (s3-0 (new 'stack-no-clear 'vector))
+                      )
+                  (vector+! s3-0 (-> arg1 trans) arg2)
+                  (cloest-point-on-mesh (-> arg0 nav) s4-0 s3-0 (the-as nav-poly #f))
+                  (cond
+                    ((< 409.6 (vector-vector-distance s3-0 s4-0))
+                     #f
+                     )
+                    ((let ((a1-4 (new 'stack-no-clear 'vector)))
+                       (set! (-> a1-4 quad) (-> s4-0 quad))
+                       (set! (-> a1-4 w) (-> arg0 root-override2 nav-radius))
+                       (add-root-sphere-to-hash! (-> arg0 nav) a1-4 #x8046e)
+                       )
+                     #f
+                     )
```

Comment:
I am reasonable certain that this function does not return a `symbol`, but it seems because it is the final form in a `cond` case it needs to return a symbol else this is a compiler error:

```
Cannot use none-typed variable in a condition.
Form:
(let ((a1-4 (new (quote stack-no-clear) (quote vector)))) (set! (-> a1-4 quad) (-> s4-0 quad)) (set! (-> a1-4 w) (-> arg0 root-override2 nav-radius)) (add-root-sphere-to-hash! (-> arg0 nav) a1-4 #x8046e))
```

Something seems wrong here